### DISCUSSION TITLE
useQuery arrayFormat configuration

### DISCUSSION THREAD

**Discussion by pranithan-kang** [TanStack/query#8017] [Q&A]
I wonder how can I send the GET request using `useQuery` method with non-square-bracket array query parameter. For example, if I have `{ foo: ["a", "b", "c"] }`, I expect the query string to be `foo=a&foo=b&foo=c` instead of `foo[]=a&foo[]=b&foo[]=c`.

**Comment by TkDodo** [comment]
react query has nothing to do with this as it doesn't make API calls. It just handles promise based state.
